src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsEventPublisher.cs

```diff
@@ -51,7 +51,41 @@ private void HandleFeatureFlagEvent(ActivityEvent activityEvent)
 
             foreach (var tag in activityEvent.Tags)
             {
-                properties[tag.Key] = tag.Value?.ToString();
+                // FeatureEvaluation event schema: https://github.com/microsoft/FeatureManagement/blob/main/Schema/FeatureEvaluationEvent/FeatureEvaluationEvent.v1.0.0.schema.json
+                if (tag.Key == "VariantAssignmentReason" && tag.Value is VariantAssignmentReason reason)
```

Comment:
```suggestion
                if (tag.Value is VariantAssignmentReason reason)
```

Doesn't seem necessary. to check for the key's value.

Reply:
I don't think we should have different serialization patterns for different VariantAssignmentReasons, or booleans.